tool/tsh/db.go

```diff
@@ -109,6 +113,95 @@ func onListDatabases(cf *CLIConf) error {
 	return trace.Wrap(showDatabases(cf.SiteName, databases, activeDatabases, roleSet, cf.Format, cf.Verbose))
 }
 
+type databaseListing struct {
+	Proxy    string           `json:"proxy"`
+	Cluster  string           `json:"cluster"`
+	roleSet  services.RoleSet `json:"-"`
+	Database types.Database   `json:"database"`
+}
+
+func listDatabasesAllClusters(cf *CLIConf) error {
+	profile, profiles, err := client.Status(cf.HomePath, "")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if profile != nil {
+		profiles = append(profiles, profile)
+	}
+
+	dbListings := make([]databaseListing, 0)
+	for _, p := range profiles {
+		cf.Proxy = p.ProxyURL.Host
```

Comment:
I would avoid overwriting proxy directly in the CLI config struct, to avoid potential unexpected side-effects later (e.g. if the code below doesn't expect it). Maybe make `makeClientWithProxy` or something?